runtime/autoload/health/provider.vim

```diff
@@ -230,10 +229,6 @@ function! s:check_python(version) abort
     if !empty(pythonx_errs)
       call health#report_warn(pythonx_errs)
     endif
-    let old_skip = get(g:, host_skip_var, 0)
-    let g:[host_skip_var] = 1
-    let [python_bin_name, pythonx_errs] = provider#pythonx#Detect(a:version)
-    let g:[host_skip_var] = old_skip
```

Comment:
If the first call to `#Detect()` didn't find any suitable Python interpreter _with_ working neovim module, why calling it again? With `g:python3_host_skip_check` set to 1, it would mean that `s:check_interpreter()` (`provider/pythonx.vim`) would only run `exepath()` (again), to get the first working Python installation _without_ working neovim module. Not sure why this would be needed. (`s:check_bin()` is almost 200 lines and no comments.)

So, using what this commit does, `:CheckHealth` would output the errors twice:

```
## Python 3 provider
  - WARNING: No Python interpreter was found with the neovim module.  Using the first available for diagnostics.
  - WARNING: provider/pythonx: Could not load Python 3:
    snowball not found in search path or not executable.
    dactyl not found in search path or not executable.
  - ERROR: Python provider error
    - SUGGESTIONS:
      - provider/pythonx: Could not load Python 3:
          snowball not found in search path or not executable.
          dactyl not found in search path or not executable.
  - INFO: Executable: Not found
```

But that's rather a problem of `:CheckHealth`, IMHO.

Alternative: `#Detect()` could also return the first working Python installation, so it needn't be called a second time with `skip_host_check` enforced.